### PR TITLE
FIX: HeapSort.java updated with new constructor with config

### LINKED ISSUE #52: Missing StatPack output for HeapSort
StatPack for Heap Sort is not logging unlike other sorts (regd. Assignment 6: Hits as time predictor).

### PR DESCRIPTION
**Pull request by AshishNevan**

### CODE REVIEW

**Comment by nehadevarapalli** [comment]
Add SortBenchmark changes

**Review by nehadevarapalli — APPROVED** [review]

File: src/main/java/edu/neu/coe/info6205/sort/elementary/HeapSort.java [hunk 1]
```diff
@@ -2,13 +2,19 @@
...
 public class HeapSort<X extends Comparable<X>> extends SortWithComparableHelper<X> {
+    public static final String DESCRIPTION = "Heap Sort";
 
     public HeapSort(Helper<X> helper) {
         super(helper);
     }
 
+    public HeapSort(int nWords, Config config) {
+        super(DESCRIPTION, nWords, 1, config);
+    }
```
Comment (glanzz) [inline src/main/java/edu/neu/coe/info6205/sort/elementary/HeapSort.java:16]:
```Java
 public HeapSort(int nWords, int nRuns, Config config) {
        super(DESCRIPTION, nWords, nRuns, config);
    }
```

File: src/main/java/edu/neu/coe/info6205/util/SortBenchmark.java [hunk 2]
```diff
@@ -174,8 +174,8 @@ void benchmarkStringSorters(String[] words, int nWords) {
             }
 
         if (isConfigBenchmarkStringSorter("heapsort") && nRunsLinearithmic > 0) {
-            Helper<String> helper = HelperFactory.create("Heapsort", nWords, config);
-            try (SortWithHelper<String> sorter = new HeapSort<>(helper)) {
+//            Helper<String> helper = HelperFactory.create("Heapsort", nWords, config);
+            try (SortWithHelper<String> sorter = new HeapSort<>(nWords, config)) {
```
Comment (glanzz) [inline src/main/java/edu/neu/coe/info6205/util/SortBenchmark.java:178]:
```Java
try (SortWithHelper<String> sorter = new HeapSort<>(nWords, nRuns, config))
```

**Review by glanzz — COMMENTED** [review]
Screenshot 2024-11-21 at 7 30 59 PM
I had fixed this by including passing <i>nRuns</i> too, else it would show up runs as 1 always in the instrumented stats logs.

File: src/main/java/edu/neu/coe/info6205/util/SortBenchmark.java (hunk 2 above)
Comment (AshishNevan) [reply src/main/java/edu/neu/coe/info6205/util/SortBenchmark.java:178]: Yes, I have observed that.
I have not made that change for 2 reasons:

1. Out of scope of this current issue thread
2. Same behavior is observed in Quick sort dual pivot, so both heapsort and quicksort need that fix.
image

**Comment by rchillyard** [comment]
OK

**Comment by rchillyard** [comment]
OK (again)

**Pull request merged**